Grid.js

```diff
@@ -0,0 +1,139 @@
+                this.cells.push(cell);
+            }        
+        }
+    }
+
+    getCellByRowCol (row, col){
+        let cell;
+        this.cells.forEach(c =>{
+            if (c.row == row && c.col == col){
+                cell = c;
+                //TODO: BREAK
+            }
+        })
+        return cell;
+    }
+    
+    assignMines(){ 
+        let numMinesCreated = 0;
+        while (numMinesCreated < this.numMines) {
+            let numGridCells = this.gridSize**2;
+            let randCell = getRandomInt(numGridCells - 1);
+            let cell = this.cells[randCell];
+            if (!cell.isMine) {
+                cell.isMine = true;
+                numMinesCreated++;
+            }
+        }
+    };
+
+    applyFnToNeighbors(cell, fn){
+        let row = cell.row;
+        let col = cell.col;
+
+        let topLeft = {row: row-1, col: col-1};
+        let topCenter = {row: row-1, col: col};
+        let topRight = {row: row-1, col: col+1};
+        
+        let centerLeft = {row: row, col: col-1};
+        let centerRight = {row: row, col: col+1};
+    
+        let bottomLeft = {row: row+1, col: col-1};
+        let bottomCenter = {row: row+1, col: col};
+        let bottomRight = {row: row+1, col: col+1};
+    
+        let neighborCoordinates = [topLeft, topCenter, topRight, 
+                                centerLeft, centerRight, 
+                                bottomLeft, bottomCenter, bottomRight];
+       
+        
+
+        const neighborCells = neighborCoordinates
+            .filter(c => this.isCellInBounds(c))
+            .map(c => this.getCellByRowCol(c.row, c.col));
+        neighborCells.forEach(neighborCell => fn(neighborCell));
+    };
+
+    isCoordinateInBounds(coordinate){
+        return coordinate >= 0 && coordinate < this.gridSize;
+    }
+
+    isCellInBounds(cell){
+        return this.isCoordinateInBounds(cell.row) && this.isCoordinateInBounds(cell.col);
+    }
+
+    isMineAndInBounds(surroundingCell) {
+        if (!this.isCellInBounds(surroundingCell)) {
+            return false;  
+        }
+        return surroundingCell.isMine;
+    }
+
+    updateThisNumMines(cell, neighborCell){
+        if(this.isMineAndInBounds(neighborCell))
+            cell.numMinesAround++;
+    }
+
+    updateEachCellNumMineAround(){
+        this.cells
+            .filter(cell => !cell.isMine)
+            .forEach(cell => {
+                cell.numMinesAround = 0;
+                this.applyFnToNeighbors(cell, neighborCell => this.updateThisNumMines(cell, neighborCell));
+        })
+    }
+
+    revealNeighbors(cell, revealedCells, zeros){
+                
+        if(!this.isCellInBounds(cell)) {
+           return;
+        }
+
+        if(cell.isRevealed){
+            return;
+        }
+
+        if (cell.numMinesAround == 0){ 
+            zeros.push(cell);
+        }
+        cell.isRevealed = true;
+        revealedCells.push(cell);
+
+    };
+
+    revealZeros  = function(cell) {
```

Comment:
the naming suggests this just performs an action, which is partly true. it also returns a list. what's the convention for function names that return something? this is also a special case, since you're returning something and updating something... feel free to use the word `and` and `or` in your function names :)